diff --git a/experiments/20260508_162317_wandb_test/run_slurm.sh b/experiments/20260508_162317_wandb_test/run_slurm.sh
index e06fd19..60fa768 100644
--- a/experiments/20260508_162317_wandb_test/run_slurm.sh
+++ b/experiments/20260508_162317_wandb_test/run_slurm.sh
@@ -69,7 +69,7 @@ apptainer exec --nv --bind "${LOCAL_SSD_DIR}" "${PROJECT_ROOT}/env/env.sif" bash
         --mouse_dataset \
         --project_path ${PROJECT_ROOT} \
         --model_name ResNet50 \
-        --ssl_name swav \
+        --ssl_name byol \
 "
 
 # 6. Sync back outputs to workspace (if local SSD was used)
diff --git a/scripts/tggate/train/train_add.py b/scripts/tggate/train/train_add.py
index 1f0f1d3..dcb467a 100755
--- a/scripts/tggate/train/train_add.py
+++ b/scripts/tggate/train/train_add.py
@@ -58,8 +58,8 @@ parser.add_argument('--ssl_name', type=str, default='barlowtwins') # ssl archite
 parser.add_argument('--num_epoch', type=int, default=50) # epoch
 parser.add_argument('--batch_size', type=int, default=64) # batch size
 parser.add_argument('--lr', type=float, default=0.01) # learning rate
-parser.add_argument('--patience', type=int, default=3) # early stopping
-parser.add_argument('--delta', type=float, default=0.002) # early stopping
+parser.add_argument('--patience', type=int, default=7) # early stopping
+parser.add_argument('--delta', type=float, default=0) # early stopping
 # Transform (augmentation) settings
 parser.add_argument('--color_plob', type=float, default=0.8)
 parser.add_argument('--blur_plob', type=float, default=0.4)
@@ -222,7 +222,7 @@ class RankMonitor:
         for x in dataloader:
             x = x.to(device)
             # DINO等の場合は、backbone(x) や model.get_intermediate_layers(x) を指定
-            z = model(x) 
+            z = model.backbone(x) 
             # もし z が (N, C, H, W) なら Global Average Pooling等でベクトル化
             if z.dim() > 2:
                 z = torch.mean(z, dim=[2, 3])
@@ -233,6 +233,7 @@ class RankMonitor:
                 break
         
         Z = torch.cat(embeddings)[:self.samples_n]
+        d = Z.size(1) # 埋め込み次元数
         
         # 2. 特異値分解 (SVD)
         # ランク崩壊の兆候を見るため、中心化（各次元の平均を引く）を行う
@@ -244,8 +245,9 @@ class RankMonitor:
         p = singular_values / torch.sum(singular_values)
         entropy = -torch.sum(p * torch.log(p))
         eff_rank = torch.exp(entropy)
+        normalized_eff_rank = eff_rank / d
         
-        return eff_rank.item(), singular_values.numpy()
+        return eff_rank.item(), normalized_eff_rank.item(), singular_values.numpy()
 
 # train epoch
 def train_epoch(model, data_loader, criterion, optimizer, epoch):
@@ -284,9 +286,10 @@ def train(model, criterion, optimizer, scheduler, early_stopping, num_epoch:int=
             f'Epoch: {epoch + 1}, train_loss: {train_epoch_loss:.4f}'
             )
         LOGGER.logger.info('elapsed_time: {:.2f} min'.format((time.time() - start)/60))
-        eff_rank, s_values = monitor.compute_effective_rank(model, val_loader, DEVICE)
+        eff_rank, normalized_eff_rank, s_values = monitor.compute_effective_rank(model, val_loader, DEVICE)
         LOGGER.logger.info(f'Effective Rank: {eff_rank:.2f}')
-        
+        LOGGER.logger.info(f'Normalized Effective Rank: {normalized_eff_rank:.2f}')
+        LOGGER.logger.info(f'Singular Values: {s_values}')
         # save model
         state = {
             "epoch":epoch,
@@ -298,7 +301,7 @@ def train(model, criterion, optimizer, scheduler, early_stopping, num_epoch:int=
             "train_loss":train_loss
         }
         if run:
-            run.log({"train_loss": train_epoch_loss, "effective_rank": eff_rank})
+            run.log({"train_loss": train_epoch_loss, "effective_rank": eff_rank, "normalized_effective_rank": normalized_eff_rank})
         torch.save(state, f'{DIR_NAME}/state.pt')
         LOGGER.save_logger(fileout=file_log)
         # state check
diff --git a/src/sslmodel/models/byol.py b/src/sslmodel/models/byol.py
index 91bfb3a..b394cfc 100755
--- a/src/sslmodel/models/byol.py
+++ b/src/sslmodel/models/byol.py
@@ -67,9 +67,9 @@ def SimSiamMLP(dim, projection_size, hidden_size=512):
 
 # Main Wrapper
 class NetWrapper(nn.Module):
-    def __init__(self, net, projection_size, projection_hidden_size, layer = -2, use_simsiam_mlp = False):
+    def __init__(self, backbone, projection_size, projection_hidden_size, layer = -2, use_simsiam_mlp = False):
         super().__init__()
-        self.net = net
+        self.backbone = backbone
         self.layer = layer
         self.projector = None
         self.projection_size = projection_size
@@ -80,10 +80,10 @@ class NetWrapper(nn.Module):
 
     def _find_layer(self):
         if type(self.layer) == str:
-            modules = dict([*self.net.named_modules()])
+            modules = dict([*self.backbone.named_modules()])
             return modules.get(self.layer, None)
         elif type(self.layer) == int:
-            children = [*self.net.children()]
+            children = [*self.backbone.children()]
             return children[self.layer]
         return None
 
@@ -109,13 +109,13 @@ class NetWrapper(nn.Module):
 
     def get_representation(self, x):
         if self.layer == -1:
-            return self.net(x)
+            return self.backbone(x)
 
         if not self.hook_registered:
             self._register_hook()
 
         self.hidden.clear()
-        _ = self.net(x)
+        _ = self.backbone(x)
         hidden = self.hidden[x.device]
         self.hidden.clear()
         return hidden
@@ -131,7 +131,7 @@ class NetWrapper(nn.Module):
 class BYOL(nn.Module):
     def __init__(
         self,
-        net,
+        backbone,
         image_size,
         hidden_layer = -2,
         projection_size = 256,
@@ -140,8 +140,8 @@ class BYOL(nn.Module):
         DEVICE=None
     ):
         super().__init__()
-        self.net = net
-        self.online_encoder = NetWrapper(net, projection_size, projection_hidden_size, layer=hidden_layer, use_simsiam_mlp=False)
+        self.backbone = backbone
+        self.online_encoder = NetWrapper(backbone, projection_size, projection_hidden_size, layer=hidden_layer, use_simsiam_mlp=False)
         self.target_encoder = None
         self.target_ema_updater = EMA(moving_average_decay)
         self.online_predictor = MLP(
diff --git a/src/sslmodel/models/simsiam.py b/src/sslmodel/models/simsiam.py
index a7fdbe7..2be9dd3 100755
--- a/src/sslmodel/models/simsiam.py
+++ b/src/sslmodel/models/simsiam.py
@@ -14,7 +14,7 @@ class SimSiam(nn.Module):
     """
     SimSiam model.
     """
-    def __init__(self, base_encoder, head_size=512, dim=2048, pred_dim=512):
+    def __init__(self, backbone, head_size=512, dim=2048, pred_dim=512):
         """
         head_size: last layer dimension
         dim: feature dimension (default: 2048)
@@ -23,7 +23,7 @@ class SimSiam(nn.Module):
         super(SimSiam, self).__init__()
 
         # set the backbone encoder
-        self.encoder = base_encoder
+        self.backbone = backbone
 
         # build a 3-layer projector
         self.fc = nn.Sequential(nn.Linear(head_size, head_size, bias=False),
@@ -53,8 +53,8 @@ class SimSiam(nn.Module):
         """
 
         # compute features for one view
-        z1 = self.encoder(x1)
-        z2 = self.encoder(x2)
+        z1 = self.backbone(x1)
+        z2 = self.backbone(x2)
         z1 = torch.flatten(z1, start_dim=1) #flatten
         z2 = torch.flatten(z2, start_dim=1) #flatten
         z1 = self.fc(z1) # NxC
diff --git a/wandb/latest-run b/wandb/latest-run
index b0f276b..3c29936 120000
--- a/wandb/latest-run
+++ b/wandb/latest-run
@@ -1 +1 @@
-run-20260509_164132-gqtitpiu
\ No newline at end of file
+run-20260509_183700-lgp0dcby
\ No newline at end of file
diff --git a/wandb/run-20260509_143926-21gzui5a/run-21gzui5a.wandb b/wandb/run-20260509_143926-21gzui5a/run-21gzui5a.wandb
index badcc05..4c42635 100644
Binary files a/wandb/run-20260509_143926-21gzui5a/run-21gzui5a.wandb and b/wandb/run-20260509_143926-21gzui5a/run-21gzui5a.wandb differ
diff --git a/wandb/run-20260509_164132-gqtitpiu/run-gqtitpiu.wandb b/wandb/run-20260509_164132-gqtitpiu/run-gqtitpiu.wandb
index 3097740..b0ca04b 100644
Binary files a/wandb/run-20260509_164132-gqtitpiu/run-gqtitpiu.wandb and b/wandb/run-20260509_164132-gqtitpiu/run-gqtitpiu.wandb differ
